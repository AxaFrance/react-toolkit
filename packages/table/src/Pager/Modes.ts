enum Modes {
  default = 'default',
  light = 'light',
}

export default Modes;
