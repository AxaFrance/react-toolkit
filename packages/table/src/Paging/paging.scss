@import '@axa-fr/react-toolkit-core/src/common/scss/core.scss';

.af-paging {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: flex-start;

  &__limit {
    order: 2;
    margin-top: 2rem;
    color: $color-pager-link;

    .af-form__group {
      margin: 0;
      font-size: 0.9375rem;

      &-label {
        font-weight: normal;
      }

      .af-form__input-select {
        padding: 2px 29px 2px 14px;
      }
    }
  }

  &__pager {
    order: 1;
    flex-basis: 100%;
  }
}

@include media-breakpoint-up(md) {
  .af-paging {
    &__limit {
      margin: 0;
      order: 1;
      flex-basis: 40%;
    }

    &__pager {
      order: 2;
      flex-basis: 60%;
    }
  }
}
@include media-breakpoint-up(xl) {
  .af-paging {
    &__limit {
      order: 1;
      flex-basis: 30%;
    }

    &__pager {
      order: 2;
      flex-basis: 70%;
    }
  }
}
