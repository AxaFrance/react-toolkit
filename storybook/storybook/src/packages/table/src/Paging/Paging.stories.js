import React from 'react';
import { Paging, Modes } from '@axa-fr/react-toolkit-table';
import Readme from '@axa-fr/react-toolkit-table/src/Paging/README.md';

export default {
  title: 'Table/Paging',
  component: Paging,
  parameters: {
    readme: {
      sidebar: Readme,
    },
    options: {},
  },
  argTypes: {
    onChange: { action: 'changed' },
    mode: {
      control: {
        type: 'select',
        options: [Modes.default, Modes.light],
      },
    },
    numberItems: {
      control: {
        type: 'select',
        options: [5, 10, 25, 50, 100],
      },
    }
  },
};

const Template = args => <Paging {...args} />;

export const Default = Template.bind({});
Default.args = {
  classModifier: '',
  className: '',
  mode: Modes.default,
  displayLabel: 'Show',
  elementsLabel: 'elements',
  ofLabel: 'of',
  nextLabel: 'Next »',
  previousLabel: '« Previous',
  currentPage: 5,
  numberPages: 23,
  numberItems: 25,
};
