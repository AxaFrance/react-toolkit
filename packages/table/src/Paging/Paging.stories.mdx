import { Meta, Story, Canvas } from '@storybook/addon-docs';
import { ArgsTable } from '@storybook/addon-docs';
import Paging from './Paging';
import Modes from '../Pager/Modes';

<Meta
  title="Components/Table/Paging/Paging"
  component={Paging}
  argTypes={{
    onChange: { action: 'changed' },
    mode: {
      control: {
        type: 'select',
        options: [Modes.default, Modes.light],
      },
    },
    numberItems: {
      control: {
        type: 'select',
        options: [5, 10, 25, 50, 100],
      },
    },
  }}
/>

# Table full paging

export const Template = (args) => <Paging {...args} />;

The Paging component uses the Items and Pager components to create a full paging experience for the user.  
Most of the properties are linked to the Items or Pager components, if you want more details on them, go to the specific documentation of the [Item component](?path=/docs/components-table-paging-items--items) or [Pager component](?path=/docs/components-table-paging-pager--pager).


Don't forget to import the component css files, so that your full paging have the correct style.

```javascript
import '@axa-fr/react-toolkit-table/dist/Paging/af-paging.css';
import '@axa-fr/react-toolkit-table/dist/Pager/af-pager.css';
import '@axa-fr/react-toolkit-form-input-select/dist/af-select.css';
import '@axa-fr/react-toolkit-form-core/dist/af-form.css';
```

<Canvas>
  <Story
    name="Paging"
    args={{
      mode: Modes.default,
      currentPage: 5,
      numberPages: 23,
    }}>
    {Template.bind({})}
  </Story>
</Canvas>
