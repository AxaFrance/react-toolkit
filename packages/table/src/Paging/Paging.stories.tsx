import React from 'react';
import { Story } from '@storybook/react';
import Paging, { PagingProps } from './Paging';
import Modes from '../Pager/Modes';
import Readme from './README.md';

export default {
  title: 'Table/Paging',
  component: Paging,
  parameters: {
    readme: {
      sidebar: Readme,
    },
    options: {},
  },
  argTypes: {
    onChange: { action: 'changed' },
    mode: {
      control: {
        type: 'select',
        options: [Modes.default, Modes.light],
      },
    },
    numberItems: {
      control: {
        type: 'select',
        options: [5, 10, 25, 50, 100],
      },
    },
  },
};

const Template: Story<PagingProps> = args => <Paging {...args} />;

export const Default = Template.bind({});
Default.args = {
  classModifier: '',
  className: '',
  mode: Modes.default,
  displayLabel: 'Show',
  elementsLabel: 'elements',
  ofLabel: 'of',
  nextLabel: 'Next »',
  previousLabel: '« Previous',
  currentPage: 5,
  numberPages: 23,
  numberItems: 25,
};
