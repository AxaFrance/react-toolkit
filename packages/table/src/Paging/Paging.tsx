import React from 'react';
import {
  withClassDefault,
  withClassModifier,
  WithClassModifierOptions,
  WithOnChangeEvent,
  OnChangeCustomEvent,
  compose,
} from '@axa-fr/react-toolkit-core';
import Pager, { PagerComponentProps } from '../Pager/Pager';
import Items, { ItemsComponentProps } from '../Items/Items';

const DEFAULT_CLASSNAME = 'af-paging';

export interface PagingEvent {
  numberItems: number;
  page: number;
}

export type PagingComponentProps = React.HTMLProps<HTMLTableElement> &
  WithOnChangeEvent<PagingEvent> &
  PagerComponentProps &
  ItemsComponentProps;

const defaultProps: Partial<PagingComponentProps> = {
  numberItems: 10,
  numberPages: 1,
  currentPage: 1,
  id: null,
};

class Paging extends React.PureComponent<PagingComponentProps> {
  constructor(props: PagingComponentProps) {
    super(props);
    this.onChangePage = this.onChangePage.bind(this);
    this.onChangeItems = this.onChangeItems.bind(this);
  }

  onChangePage(data: OnChangeCustomEvent) {
    const { onChange, numberItems } = this.props;
    onChange({
      numberItems,
      page: data.value,
    });
  }

  onChangeItems(data: OnChangeCustomEvent) {
    const { onChange, currentPage } = this.props;
    onChange({
      numberItems: data.value,
      page: currentPage,
    });
  }

  render() {
    const {
      className,
      currentPage,
      displayLabel,
      elementsLabel,
      id,
      mode,
      nextLabel,
      numberItems,
      numberPages,
      ofLabel,
      previousLabel,
    } = this.props;

    return (
      <div className={className}>
        <div className="af-paging__limit">
          <Items
            onChange={this.onChangeItems}
            numberItems={numberItems}
            id={id}
            displayLabel={displayLabel}
            elementsLabel={elementsLabel}
          />
        </div>
        <div className="af-paging__pager">
          <Pager
            onChange={this.onChangePage}
            currentPage={currentPage}
            numberPages={numberPages}
            previousLabel={previousLabel}
            nextLabel={nextLabel}
            ofLabel={ofLabel}
            mode={mode}
          />
        </div>
      </div>
    );
  }
}

export type PagingProps = PagingComponentProps & WithClassModifierOptions;

const enhance = compose<PagingComponentProps, PagingProps>(
  withClassDefault(DEFAULT_CLASSNAME),
  withClassModifier
);
const Enhance = enhance(Paging);
Enhance.defaultProps = defaultProps;

export default Enhance;
