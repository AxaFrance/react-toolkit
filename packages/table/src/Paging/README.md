## Paging

### Installation

```sh
npm i @axa-fr/react-toolkit-table
```

### Import

```javascript
import { Paging } from '@axa-fr/react-toolkit-table';

import '@axa-fr/react-toolkit-table/dist/Paging/paging.scss';
/* or */
import '@axa-fr/react-toolkit-table/dist/Paging/af-paging.css';
```
