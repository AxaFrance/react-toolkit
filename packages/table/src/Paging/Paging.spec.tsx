import React from 'react';
import { create } from 'react-test-renderer';
import Paging from './Paging';
import Modes from '../Pager/Modes';

describe('Table.Paging', () => {
  it('renders correctly with default values', () => {
    const handleOnChangeMock = jest.fn();
    handleOnChangeMock.mockImplementationOnce(() => 'My mock is called');

    const component = create(
      <Paging
        numberItems={20}
        numberPages={10}
        onChange={handleOnChangeMock}
        currentPage={1}
        id="pagingid"
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with custom values', () => {
    const handleOnChangeMock = jest.fn();
    handleOnChangeMock.mockImplementationOnce(() => 'My mock is called');

    const component = create(
      <Paging
        numberItems={20}
        numberPages={10}
        onChange={handleOnChangeMock}
        currentPage={1}
        id="pagingid"
        previousLabel="« Previous"
        nextLabel="Next »"
        ofLabel="of"
        displayLabel="Show"
        elementsLabel="elements"
        mode={Modes.light}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
