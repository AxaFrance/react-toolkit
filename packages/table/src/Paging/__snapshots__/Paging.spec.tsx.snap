// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.Paging renders correctly 1`] = `
<div
  className="af-paging"
>
  <div
    className="af-paging__limit"
  >
    <div
      className="af-paging__limit"
    >
      <form
        className="af-paging__form"
      >
        <div
          className="af-form__group"
        >
          <div
            className="col col-sm-2 col-md-2 col-lg-2 col-xl-2"
          >
            <label
              className="af-form__group-label"
              htmlFor="pagingid"
            >
              Afficher
            </label>
          </div>
          <div
            className="col col-sm-10 col-md-10 col-lg-10 col-xl-10"
          >
            <div
              className="af-form__select"
            >
              <div
                className="af-form__select-container"
              >
                <select
                  className="af-form__input-select"
                  id="pagingid"
                  onChange={[Function]}
                  value={20}
                >
                  <option
                    value={5}
                  >
                    5
                  </option>
                  <option
                    value={10}
                  >
                    10
                  </option>
                  <option
                    value={25}
                  >
                    25
                  </option>
                  <option
                    value={50}
                  >
                    50
                  </option>
                  <option
                    value={100}
                  >
                    100
                  </option>
                </select>
                <span
                  className="glyphicon glyphicon-menu-down"
                />
              </div>
              <span
                className="af-form__input-cmplt"
              >
                éléments
              </span>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    className="af-paging__pager"
  >
    <nav
      className="af-pager"
    >
      <ul
        className="af-pager__pagination"
      >
        <li
          className="af-pager__item af-pager__item--disabled"
        >
          <span
            className="af-pager__item-nolink"
          >
            « Précédent
          </span>
        </li>
        <li
          className="af-pager__item af-pager__item--active"
        >
          <a
            className="af-pager__item-link"
            href="#"
            onClick={[Function]}
          >
            <span>
              1
            </span>
          </a>
        </li>
        <li
          className="af-pager__item"
        >
          <a
            className="af-pager__item-link"
            href="#"
            onClick={[Function]}
          >
            <span>
              2
            </span>
          </a>
        </li>
        <li
          className="af-pager__item af-pager__item--disabled"
        >
          <span
            className="af-pager__item-link"
          >
            ...
          </span>
        </li>
        <li
          className="af-pager__item"
        >
          <a
            className="af-pager__item-link"
            href="#"
            onClick={[Function]}
          >
            <span>
              10
            </span>
          </a>
        </li>
        <li
          className="af-pager__item"
        >
          <a
            className="af-pager__item-link"
            href="#"
            onClick={[Function]}
            role="button"
          >
            Suivant »
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
